.github/workflows/bump-deps.yaml

```diff
@@ -0,0 +1,63 @@
+name: Bump DEPS
+on:
+  workflow_dispatch:
+    inputs:
+      dep:
+        description: "Dependency"
+        required: true
+        type: string
+        default: "vendor/web-discovery-project"
+      ref:
+        description: "New ref for dependency"
+        required: true
+        type: string
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-bump:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: auto-bump
+        run: |
+          set -x
+
+          git config user.email "${{ github.actor }}@users.noreply.github.com"
+          git config user.name "${{ github.actor }}"
+          git config push.default simple
+
+          SED=sed
+          DATE=date
+
+          # Inputs
+          DEP="${{ inputs.dep }}"
+          DEPREF="${{ inputs.ref }}"
+          [[ -z "$DEP" ]] && { echo Input dependency cannot be empty; exit 1; }
+          [[ -z "$DEPREF" ]] && { echo Input ref cannot be empty; exit 1; }
+
+          DATETIME="$($DATE '+%Y-%m-%d-%H-%M')"
+          readarray -t milestones < <(gh api /repos/brave/brave-core/milestones -q ".[] | .title" | grep -Fv Closed)
+          branches=( "${milestones[@]/*Nightly/master}" )
+          branches=( "${branches[@]/ -*/}" )
+          echo Branches to operate on:
+          declare -p branches
+          for branch in "${branches[@]}"
+          do
+            git switch "$branch"
+            pr_branch="pr-auto-bump-deps-${branch}-${DATETIME}"
+            git switch -c "$pr_branch"
+            "$SED" -i -e "s|\(.*${DEP}.*\)@\(.*\)\",$|\1@${DEPREF}\",|" DEPS
```

Comment:
While this is pleasantly straightforward, it wouldn't work with some of the currently defined deps, like vendor/omaha for instance.

You could use something along the lines of
```bash
sed -i -e '/^\s*"'"${DEP//\//\\\/}"'"\s*:/,/@/ s/@[^"]*"/@'"$DEPREF"'"/' DEPS
```
This applies the transformation only to lines between the two matches (the dep name and an `@` character). The syntax is `/from/,/to/ s/a/b/`. This might be GNU sed specific, but we run this on Linux anyway, so it doesn't matter.

Feel free to adjust as needed, this is just to illustrate the approach.

Reply:
Thank you @wknapik for this quoting kung-fu 👏 That is helpful!

Reply:
Yeah, it's pretty ugly ;]  The most interesting thing to note here is that you can do
```bash
echo "\""
```
to print a double quote, but you can't do
```bash
echo '\''
```
to print a single quote, so the easiest way to print a single quote is from a double-quoted string
```bash
echo "'"
```
and you can combine single- and double-quoted strings as needed, so to print `'"`, you can do
```bash
echo "'"'"'
```

Reply:
```bash
sed -e "s,^\(\s*\"$DEP\"\s*:\s*\"https.*\)@[^\"]*\",\1@$DEPREF," \
    -e "/^\s*\"${DEP//\//\\/}\"\s*:\s*{/,/@/ s/@[^\"]*/@$DEPREF/" DEPS
```
This would be sensitive to some newline changes in DEPS, but should correctly cover both the single- and multi-line cases.

Reply:
It's too bad DEPS aren't valid json, even if you extract what's assigned to `deps` - would be easier to manipulate safely - but that's how it is, so we have to make do ;]